src/main/java/org/opensearch/flowframework/common/FlowFrameworkMaxRequestRetrySetting.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.common;
+
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+
+import static org.opensearch.flowframework.common.FlowFrameworkSettings.MAX_REQUEST_RETRY;
+
+/**
+ * Controls max request retry setting for workflow step transport action APIs
+ */
+public class FlowFrameworkMaxRequestRetrySetting {
```

Comment:
Instead of making a class for Settings. We should make an abstract class which implements `WorkflowStep` and then extends this abstract class in `GetMLTaskStep`. This way whenever `GetMLTaskStep` is invoked the constructor of the abstract class will create these settings. Ref: `AbstractWorkflowAction.java`